server.go

```diff
@@ -67,7 +68,9 @@ func (svr *Server) getHandle(handle string) (*os.File, bool) {
 
 type serverRespondablePacket interface {
 	encoding.BinaryUnmarshaler
+	id() uint32
 	respond(svr *Server) error
+	writing() bool
```

Comment:
would `readonly` be a better name for this ? it does have a problem with the zero value making r/w the default, but writing isn't a very good name for a method IMO